src/main.rs

```diff
@@ -1,44 +1,80 @@
-use clap::{App, Arg};
+use clap::{crate_authors, crate_version, App, AppSettings, Arg};
+use clap_generate::generators::*;
+use clap_generate::{generate, Generator};
 use std::env;
+use std::io;
 
 mod cli;
 mod git;
 
-fn main() {
-    let matches = App::new("git local-ignore")
+// cli interface
+fn cli_gl() -> App<'static> {
+    App::new("git-local-ignore")
         .version(env!("CARGO_PKG_VERSION"))
         .author("Vyacheslav P. <vyacheslav.pukhanov@gmail.com>")
```

Comment:
We could probably replace these with the new `crate_version` and `crate_authors` macros? They weren't available before

Reply:
This is definitely an option, but I did not want to mess up too much with your code.